lib/msf/core/payload/python/meterpreter_loader.rb

```diff
@@ -264,28 +262,46 @@ def _s2b(text):
 		return[ord(c)for c in text]
 	def _b2s(binary):
 		return bytes(binary)
-	def _concat_list(a,b):
-		return a+bytes(b)
 else:
 	bytes=lambda s,e:s
+def _gmul(a, b):
+	r=0
+	while b:
+		if b&1: r^=a
+		a<<=1
+		if a>255: a^=0x11B
+		b>>=1
+	return r
+def _mix(n, vec):
+	return sum([_gmul(n,v)<<(24-8*shift) for shift,v in enumerate(vec)])
+def _ror32(n):
+	return (n&255)<<24|n>>8
+def _rcon():
+	return [_gmul(1, 1<<n) for n in xrange(30)]
+def _Si(S):
+	return [S.index(n) for n in xrange(len(S))]
+def _mixl(S, vec):
+	return [_mix(s, vec) for s in S]
+def _rorl(T):
+	return [_ror32(t) for t in T]
 empty=struct.pack('')
 class AESCBC(object):
 	nrs={16:10,24:12,32:14}
-	rcon=[1,2,4,8,16,32,64,128,27,54,108,216,171,77,154,47,94,188,99,198,151,53,106,212,179,125,250,239,197,145]
+	rcon=_rcon()
```

Comment:
This is the only place where `_rcon()` is used, you might want to inline it.

Reply:
Unfortunately, on Python 3 it will result in a NameError. https://stackoverflow.com/q/13905741/477563 ☹️ 
It leads to rather goofy looking code but the current approach is the only one that's Python 2/3 portable without moving the calculation to the `__init__` method.

Reply:
`rcon = [_gmul(1, 1<<n) for n in range(30)]` doesn't access any class variable